test/ntttcp_test.py

```diff
@@ -0,0 +1,470 @@
+        parse_result = NtttcpResult(result.rec_stdout, result.send_stdout)
+        assert parse_result.is_output_xml_file(filename)
+#       assert throughput >= self.expected_throughput * 0.9
+
+    def test_ouput_json_file(self) -> None:
+        filename = "myjson.json"
+        sender_option = f"-j {filename}"
+        rec_cmd, send_cmd = self.combine_command(
+                sender_option=sender_option
+        )
+        result = self.run_test(rec_cmd, send_cmd)
+        parse_result = NtttcpResult(result.rec_stdout, result.send_stdout)
+        assert parse_result.is_output_json_file(filename)
+#        assert throughput >= self.expected_throughput * 0.9
+
+    def test_starting_port_number(self) -> None:
+        starting_port = 10000
+        common_option = f"-p {starting_port}"
+        sender_option = "-V"
+        rec_cmd, send_cmd = self.combine_command(
+                common_option=common_option,
+                sender_option=sender_option
+        )
+        result = self.run_test(rec_cmd, send_cmd)
+        parse_result = NtttcpResult(result.rec_stdout, result.send_stdout)
+        assert parse_result.get_starting_port_number() == starting_port
+        throughput = parse_result.get_throughput()
+        print("throughput =", throughput)
+
+    def test_mapping_option(self) -> None:
+        ports = 200
+        defualt_threads = 4
+        rec_cmd = f"ulimit -n 10240 && ntttcp -D -r -m {ports},*,127.0.0.1 -D -t 5"
+        send_cmd = f"ulimit -n 10240 && ntttcp -s127.0.0.1 -P {ports} -t 5"
+        result = self.run_test(rec_cmd, send_cmd)
+        parse_result = NtttcpResult(result.rec_stdout, result.send_stdout)
+        assert parse_result.get_ports_numbers() == ports * defualt_threads
+        throughput = parse_result.get_throughput()
+        print("throughput =", throughput)
+
+    def test_system_tcp_retransmit(self) -> None:
+        common_option = "--show-tcp-retrans"
+        rec_cmd, send_cmd = self.combine_command(
+                common_option=common_option
+        )
+        result = self.run_test(rec_cmd, send_cmd)
+        parse_result = NtttcpResult(result.rec_stdout, result.send_stdout)
+        assert parse_result.is_having_system_tcp_retransmit() is True
+        throughput = parse_result.get_throughput()
+        print("throughput =", throughput)
+
+    def test_show_nic_packets(self) -> None:
+        common_option = "--show-nic-packets lo"
+        rec_cmd, send_cmd = self.combine_command(
+                common_option=common_option,
+        )
+        result = self.run_test(rec_cmd, send_cmd)
+        parse_result = NtttcpResult(result.rec_stdout, result.send_stdout)
+        assert parse_result.is_show_nic_packets() is True
+        throughput = parse_result.get_throughput()
+        print("throughput =", throughput)
+
+    def test_show_dev_interrupts(self, ) -> None:
+        common_option = "--show-dev-interrupts Hypervisor callback interrupts"
```

Comment:
Do we need to quota the vector name like:
"--show-dev-interrupts 'Hypervisor callback interrupts'"

Reply:
done